snakemake_executor_plugin_latch/resource_utils.py

```diff
@@ -0,0 +1,216 @@
+                f"{format_size(self.disk)} Storage",
+                *(
+                    [f"{self.gpus} {self.gpu_type} GPU(s)"]
+                    if self.gpus > 0 and self.gpu_type is not None
+                    else []
+                ),
+            ]
+        )
+
+    def __eq__(self, other):
+        if not isinstance(other, Resources):
+            raise TypeError(
+                f"'=' not supported between instances of '{type(self)}' and '{type(other)}'"
+            )
+
+        return (
+            self.cpu == other.cpu
+            and self.mem == other.mem
+            and self.disk == other.disk
+            and self.gpus == other.gpus
+            and self.gpu_type == other.gpu_type
+        )
+
+    def __le__(self, other):
+        if not isinstance(other, Resources):
+            raise TypeError(
+                f"'<' not supported between instances of '{type(self)}' and '{type(other)}'"
+            )
+
+        return self.cpu < other.cpu and self.mem < other.mem and self.disk < other.disk
+
+    def __leq__(self, other):
+        return self.__eq__(other) or self.__le__(other)
+
+    def __ge__(self, other):
+        return not self.__leq__(other)
+
+    def __geq__(self, other):
+        return not self.__le__(other)
+
+
+_ng_limits = {
+    "gpu-small": Resources(
+        cpu=7_000,
+        mem=parse_size("30 GiB"),
+        disk=parse_size("1500 GiB"),
+        gpus=1,
+        gpu_type="nvidia-t4",
+    ),
+    "gpu-big": Resources(
+        cpu=64_000,
+        mem=parse_size("240 GiB"),
+        disk=parse_size("5000 GiB"),
+        gpus=1,
+        gpu_type="nvidia-a10g",
+    ),
+    "v100-x1": Resources(
+        cpu=7_000,
+        mem=parse_size("48 GiB"),
+        disk=parse_size("2000 GiB"),
+        gpus=1,
+        gpu_type="nvidia-v100",
+    ),
+    "v100-x4": Resources(
+        cpu=30_000,
+        mem=parse_size("230 GiB"),
+        disk=parse_size("2000 GiB"),
+        gpus=4,
+        gpu_type="nvidia-v100",
+    ),
+    "v100-x8": Resources(
+        cpu=62_000,
+        mem=parse_size("400 GiB"),
+        disk=parse_size("2000 GiB"),
+        gpus=8,
+        gpu_type="nvidia-v100",
+    ),
+    "cpu-32-spot": Resources(
+        cpu=30_000,
+        mem=parse_size("120 GiB"),
+        disk=parse_size("2000 GiB"),
+    ),
+    "cpu-96-spot": Resources(
+        cpu=94_000,
+        mem=parse_size("176 GiB"),
+        disk=parse_size("4949 GiB"),
+    ),
+    "mem-512-spot": Resources(
+        cpu=62_000,
+        mem=parse_size("485 GiB"),
+        disk=parse_size("4949 GiB"),
+    ),
+    "mem-1tb": Resources(
+        cpu=126_000,
+        mem=parse_size("975 GiB"),
+        disk=parse_size("4949 GiB"),
+    ),
+}
+
+
+_resource_key_expr = re.compile(r"^(?P<type>mem|disk)(?:_(?P<unit>\w+))?$")
+
+
+def get_resources(resources: dict[str, str]) -> Resources:
+    res = Resources()
+
+    for key, val in resources.items():
+        if key.startswith("cpu"):
```

Comment:
why check startswith instead of equality?

Reply:
i wanna support both `"cpu"` and `"cpus"` since ive seen people use them interchangeably